node/src/types/block.rs

```diff
@@ -1265,51 +1265,25 @@ impl Item for Block {
 
 /// A wrapper around `Block` for the purposes of fetching blocks by height in linear chain.
 #[derive(Clone, Debug, PartialOrd, Ord, PartialEq, Eq, Hash, Serialize, Deserialize)]
-pub enum BlockByHeight {
-    Absent(u64),
-    Block(Box<Block>),
+pub struct BlockWithMetadata {
+    pub block: Block,
+    pub finality_signatures: BlockSignatures,
 }
```

Comment:
We want to keep the previous functionality of being able to signal that block was not found in the storage. This way we're significantly improving the performance of the sync process – nodes requesting the block don't have to wait until they time out, they will be notified that the requested node did not have the block and they will continue to a next peer.